cpp/src/arrow/acero/CMakeLists.txt

```diff
@@ -173,13 +173,8 @@ add_arrow_acero_test(hash_join_node_test SOURCES hash_join_node_test.cc
                      bloom_filter_test.cc)
 add_arrow_acero_test(pivot_longer_node_test SOURCES pivot_longer_node_test.cc)
 
-# asof_join_node and sorted_merge_node use std::thread internally
-# and doesn't use ThreadPool so it will
-# be broken if threading is turned off
-if(ARROW_ENABLE_THREADING)
-  add_arrow_acero_test(asof_join_node_test SOURCES asof_join_node_test.cc)
-  add_arrow_acero_test(sorted_merge_node_test SOURCES sorted_merge_node_test.cc)
-endif()
+add_arrow_acero_test(asof_join_node_test SOURCES asof_join_node_test.cc)
+add_arrow_acero_test(sorted_merge_node_test SOURCES sorted_merge_node_test.cc)
```

Comment:
It seems that this is #41125. Let's work on #41125 before this.

Reply:
#41125 is in! This branch likely needs a refresh,but all related testing and threading issues seem to be addressed!

Thanks @joemarshall for your continued work here! Home stretch!

@kou can you do an initial review or would you rather wait until Joe does one more pass with upstream changes completed?

Reply:
> It seems that this change is #41125. Let's work on #41125 before this PR.

#41125 is in! This branch likely needs a refresh,but all related testing and threading issues seem to be addressed!

Thanks @joemarshall for your continued work here! Home stretch!

@kou can you do an initial review or would you rather wait until Joe does one more pass with upstream changes completed?

Reply:
Could you rebase on main to remove needless changes from this PR?
It'll be easy to review.